### ISSUE TITLE
test results differ on different batch sizes 

### ISSUE THREAD

**Issue by ofrimasad** [ultralytics/yolov5#1426]
## ❔Question
I am running the test.py with the exact same parameter except for the batch size and I am getting different results. 
this is not always noticeable when looking only on the final mAP results, but it seems that the model itself output a slightly different results when working with different batch sizes. 

i.e. 
testing with COCO80 and batch sizes 10 and 30, the second images in the validation set gets different number of detection after the nms initial threshold (which shows that some of the detection for that image got different confidence score)

I have made sure all of the network components are in  training=False

Do you have any cross batch normalization other than the BatchNorm2d(which is set to training=False)?  

Thanks

**Comment by glenn-jocher** [comment]
@ofrimasad different settings will naturally need to different results, I'd be worried if that was not the case.

Batch padding varies by the content images within the batch. Did you expect the batch shape to be exactly the same as you throw more and more images of different sizes into it?
